### PR TITLE
[branch-3.0](binlog) Support drop view binlog (#39781)

### PR DESCRIPTION
**Pull request by wyxxxcat**
pick #39781

### CODE REVIEW

**Comment by dataroaring** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java [hunk 1]
```diff
@@ -2127,29 +2127,25 @@ private Status allTabletCommitted(boolean isReplay) {
...
+                }
+                if (tableType == TableType.OLAP && backupTableInfo != null) {
+                    // drop the non restored partitions.
+                    dropNonRestoredPartitions(db, (OlapTable) table, backupTableInfo);
+                } else if (isCleanTables) {
+                    // otherwise drop the entire table.
+                    LOG.info("drop non restored table {}({}). {}", tableName, tableId, this);
+                    boolean isView = false;
+                    boolean isForceDrop = false; // move this table into recyclebin.
+                    env.getInternalCatalog().dropTableWithoutCheck(db, table, isView, isForceDrop);
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java:2148]: @wyxxxcat This patch is staled, please update.

**Review by w41ter — CHANGES_REQUESTED** [review]

**Comment by w41ter** [comment]
close in favor #44112

**Pull request closed without merging**